### PR TITLE
Refactor packaging and imports

### PR DESCRIPTION
**Pull request by jackrobison**
Continuing https://github.com/lbryio/lbryum/pull/111, https://github.com/lbryio/lbryum/pull/112, and https://github.com/lbryio/lbryum/pull/113... The remaining removals are consolidated in https://github.com/lbryio/lbryum/pull/121. This PR updates the module packaging and imports and adds pylinting.

before 
before

after 
after

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,8 +1,6 @@
 ecdsa==0.13
 pbkdf2
 requests
-protobuf==3.2.0
```
Comment (kaykurokawa) [inline requirements.txt]: I think this is required for lbryschema 
requiremens.txt is not recursive

Comment (jackrobison) [inline requirements.txt]: It was needed explicitly in lbryum for the payment requests. The protobuf requirement is still in the lbryschema requirements.txt

**Comment by kaykurokawa** [comment]
You have to break this up. This is just too many commits to go through in a single review. 
Something like a PR for removals, PR for pylints, and PR for any significant code changes.

**Comment by jackrobison** [comment]
@kaykurokawa I consolidated the removals to https://github.com/lbryio/lbryum/pull/121 and reorganized the remaining commits a bit better

**Comment by lyoshenka** [comment]
Is pep8 styling automated?

**Comment by jackrobison** [comment]
@lyoshenka 
You can configure pycharm to complain at you for breaking it, and it can reformat/clean up files.

**Comment by kaykurokawa** [comment]
requirements.txt should contain indirect dependencies.

please read up on: 

https://blog.miguelgrinberg.com/post/the-package-dependency-blues
https://packaging.python.org/discussions/install-requires-vs-requirements/

Thus they need protobuf which was deleted because lbryschema uses it, along with jsonschema and ecdsa 

protobuf==3.2.0
jsonschema==2.5.1
ecdsa==0.13

**Comment by jackrobison** [comment]
Added protobuf and jsonschema, ecdsa is already in there

**Comment by kaykurokawa** [comment]
please verify this PR passes both lbryum_testing.py and integration_testing.py in lbry-in-a-box before merging. otherwise LGTM

**Comment by jackrobison** [comment]
https://travis-ci.org/lbryio/lbry-in-a-box/builds/247717897

**Pull request merged**